src/Workspaces/Remote/ServiceHub/Services/SnapshotService.JsonRpcAssetSource.cs

```diff
@@ -30,20 +30,20 @@ public JsonRpcAssetSource(SnapshotService owner) : base(owner.AssetStorage)
                 _owner = owner;
             }
 
-            public override async Task<IList<(Checksum, object)>> RequestAssetsAsync(int scopeId, ISet<Checksum> checksums, ISerializerService serializerService, CancellationToken callerCancellation)
+            public override async Task<IList<(Checksum, object)>> RequestAssetsAsync(int scopeId, ISet<Checksum> checksums, ISerializerService serializerService, CancellationToken cancellationToken)
             {
-                using (RoslynLogger.LogBlock(FunctionId.SnapshotService_RequestAssetAsync, GetRequestLogInfo, scopeId, checksums, callerCancellation))
+                using (RoslynLogger.LogBlock(FunctionId.SnapshotService_RequestAssetAsync, GetRequestLogInfo, scopeId, checksums, cancellationToken))
                 {
                     try
                     {
-                        return await _owner.RunServiceAsync(cancellationToken =>
+                        return await _owner.RunServiceAsync(() =>
                         {
                             return _owner.Rpc.InvokeAsync(WellKnownServiceHubServices.AssetService_RequestAssetAsync,
                                 new object[] { scopeId, checksums.ToArray() },
                                 (s, c) => ReadAssets(s, scopeId, checksums, serializerService, c), cancellationToken);
-                        }, callerCancellation).ConfigureAwait(false);
+                        }, cancellationToken).ConfigureAwait(false);
                     }
-                    catch (Exception ex) when (ReportUnlessCanceled(ex, callerCancellation))
+                    catch (Exception ex) when (ReportUnlessCanceled(ex, cancellationToken))
```

Comment:
❓ Why are we wrapping the `InvokeAsync` call in 1) `LogUnlessCanceled`, 2) `ReportUnlessCanceled`, _and_ 3) `LogBlock`?

Reply:
I want to crash OOP on this case. if this fails, OOP is in a state where it can't recover or perform. so it catch exception and crash OOP. I want to have dump for this case. 

it will make VS to show info bar saying OOP is gone but NOT crash VS.